packages/BiostructureViewer/src/utils/ngl-gl-doc-service.ts

```diff
@@ -0,0 +1,210 @@
+import * as grok from 'datagrok-api/grok';
+import * as ui from 'datagrok-api/ui';
+import * as DG from 'datagrok-api/dg';
+
+import {_package} from '../package';
+import * as NGL from 'NGL';
+import {NglGlServiceBase, NglGlTask} from '@datagrok-libraries/bio/src/viewers/ngl-gl-viewer';
+
+
+export class NglGlDocService implements NglGlServiceBase {
+  private readonly nglDiv: HTMLDivElement;
+  private readonly ngl: any;
+
+  private readonly hostDiv: HTMLDivElement | null = null;
+
+  private readonly _queue: { key?: keyof any, task: NglGlTask, dt: number }[];
+  private readonly _queueDict: { [key: keyof any]: NglGlTask };
+  /** The flag allows {@link _processQueue}() on add item to the queue with {@link render}() */
+  private _busy: boolean = false;
+
+  constructor() {
+    const r = window.devicePixelRatio;
+
+    this.nglDiv = ui.div([], 'd4-ngl-viewer');
+    this.ngl = new NGL.Stage(this.nglDiv);
+    this.ngl.viewer.signals.rendered.add(this.onNglRendered.bind(this));
+
+    // The single NGL component
+    this.hostDiv = ui.div([this.nglDiv]);
+    this.hostDiv.style.position = 'absolute';
```

Comment:
Consider using css styles. Consider hiding it by setting display to none.

Reply:
style.display='none' disables NGL drawing at all

css: There is a single use of these settings "hiding" the div. Do you insist on using css style?